core/src/test/java/org/apache/calcite/util/UtilTest.java

```diff
@@ -967,6 +968,17 @@ private String toString(byte[] bytes) {
     assertEquals(5825, x);
   }
 
+  /**
+   * Unit test for {@link Pair#zip(java.util.List, java.util.List, boolean)}.
+   */
+  @Test void testPairZipWithStrict() {
+    List<String> strings = Arrays.asList("paul", "george", "john", "ringo");
+    List<Integer> integers = Arrays.asList(1942, 1943);
+    assertThrows(IllegalArgumentException.class, () -> {
+      List<Pair<String, Integer>> zip = Pair.zip(strings, integers, true);
+    });
```

Comment:
Maybe we should hold the return value of `assertThrows`, and asserts the error message is as expected?